### PR TITLE
feat(sourceprocessor): add support for container-level annotations

### PR DESCRIPTION
**Pull request by astencel-sumo**

### CODE REVIEW

File: pkg/processor/sourceprocessor/source_category_filler_test.go [hunk 1]
```diff
@@ -62,3 +62,73 @@ func TestFillWithAnnotations(t *testing.T) {
...
+		cfg := createDefaultConfig().(*Config)
+		cfg.SourceCategory = "my-source-category"
+		cfg.ContainerAnnotations.Enabled = true
+
+		attrs := pdata.NewAttributeMap()
+		attrs.InsertString("k8s.pod.annotation.sumologic.com/container-name-1.sourceCategory", "first_source-category")
+		attrs.InsertString("k8s.pod.annotation.sumologic.com/container-name-2.sourceCategory", "another/source-category")
+		attrs.InsertString("k8s.container.name", "container-name-1")
+
+		filler := newSourceCategoryFiller("_sourceCategory", cfg)
```
Comment (pmalek-sumo) [inline pkg/processor/sourceprocessor/source_category_filler_test.go:92]: One nit: `newSourceCategoryFiller` seems to be configurable but is always called with `"_sourceCategory"` or `sourceCategoryKey` as the first param + the names says it all.

I believe we can drop this param and just hardcode it, no?

File: pkg/processor/sourceprocessor/README.md [hunk 2]
```diff
@@ -84,3 +84,35 @@ prefixed with the value defined in `keys.annotation_prefix` config option.
 This can be achieved with the [Kubernetes processor](../k8sprocessor).
 
 For example, if a resource has the `k8s.pod.annotation.sumologic.com/exclude` attribute, the resource will be dropped.
+
+### Container-level pod annotations
+
+To make it possible to set different metadata on logs from different containers inside a pod,
```
Comment (pmalek-sumo) [inline pkg/processor/sourceprocessor/README.md:90]: Do we mention it somewhere in the README what's the precedence between pod and container-level annotations?

**Review by pmalek-sumo — COMMENTED** [review]
One small nit

File: pkg/processor/sourceprocessor/README.md [hunk 3]
```diff
@@ -84,3 +84,35 @@ prefixed with the value defined in `keys.annotation_prefix` config option.
 This can be achieved with the [Kubernetes processor](../k8sprocessor).
 
 For example, if a resource has the `k8s.pod.annotation.sumologic.com/exclude` attribute, the resource will be dropped.
+
+### Container-level pod annotations
```
Comment (sumo-drosiek) [inline pkg/processor/sourceprocessor/README.md:88]: Do we support source category prefix for container-level pod annotations?

File: pkg/processor/sourceprocessor/source_category_filler_test.go (hunk 1 above)
Comment (sumo-drosiek) [reply pkg/processor/sourceprocessor/source_category_filler_test.go:92]: in addition we can hardcode all `_source` headers as we treat it special way in exporter

File: pkg/processor/sourceprocessor/README.md (hunk 3 above)
Comment (astencel-sumo) [reply pkg/processor/sourceprocessor/README.md:88]: Later in the section:

```
No other transformations are applied to the source categories retrieved from container-level annotations, like adding source category prefix or replacing the dash.
```

File: pkg/processor/sourceprocessor/README.md (hunk 2 above)
Comment (astencel-sumo) [reply pkg/processor/sourceprocessor/README.md:90]: Added note https://github.com/SumoLogic/sumologic-otel-collector/pull/260/commits/3acb6e8688cff8f956f2bb9a33d39b827f9e53a9

File: pkg/processor/sourceprocessor/source_category_filler_test.go (hunk 1 above)
Comment (astencel-sumo) [reply pkg/processor/sourceprocessor/source_category_filler_test.go]: Removed `attributeName` parameter in https://github.com/SumoLogic/sumologic-otel-collector/pull/260/commits/375315674010f0e344d265c7ff5d0cbb781a4ab1

File: pkg/processor/sourceprocessor/README.md [hunk 4]
```diff
@@ -84,3 +84,36 @@ prefixed with the value defined in `keys.annotation_prefix` config option.
...
+
+To make it possible to set different metadata on logs from different containers inside a pod,
+it is possible to set pod annotations that are container-specific.
+Container-level annotations take precendence over other forms of setting the source category.
```
Comment (sumo-drosiek) [inline pkg/processor/sourceprocessor/README.md:92]: can we bold it?

File: pkg/processor/sourceprocessor/README.md [hunk 5]
```diff
@@ -84,3 +84,36 @@ prefixed with the value defined in `keys.annotation_prefix` config option.
...
+setting the following annotations on the pod:
+
+- `sumologic.com/container-name-1.sourceCategory` with the value of `first_source-category`
+- `sumologic.com/container-name-2.sourceCategory` with the value of `another/source-category`
+
+will make the logs from `container-name-1` be tagged with source category `first_source-category`
+and logs from `container-name-2` be tagged with source category `another/source-category`.
+
+No other transformations are applied to the source categories retrieved from container-level annotations,
+like adding source category prefix or replacing the dash.
```
Comment (sumo-drosiek) [inline pkg/processor/sourceprocessor/README.md:115]: I would bold it too

File: pkg/processor/sourceprocessor/README.md (hunk 4 above)
Comment (astencel-sumo) [reply pkg/processor/sourceprocessor/README.md]: I'm not a fan of bolding half of the docs ;)
So I tried the make the most relevant points more visible moving them into a list near the top of the docs. How about this? https://github.com/SumoLogic/sumologic-otel-collector/pull/260/commits/0cff98d39e82c7df535f91275dc91c0439ce30b8

Comment (sumo-drosiek) [reply pkg/processor/sourceprocessor/README.md]: It's better. It's little but harder to miss it while reading

**Review by pmalek-sumo — APPROVED** [review]

**Comment by pmalek-sumo** [comment]
Please resolve conflicts 😄

**Pull request merged**